src/update.c

```diff
@@ -367,6 +367,55 @@ static int main_update()
 	retries = 0;
 	timeout = 10;
 
+	/* Get a list of all the iterative manifests from the server_manifest that match
+	* the "from" version, also get a list of all the full manifests in the MoM */
+	struct list *iterative_manifests = NULL;
+	struct list *full_manifests = NULL;
+	struct list *list1;
+	struct file *file1;
+
+	list1 = server_manifest->manifests;
+	while (list1) {
+		file1 = list1->data;
+		list1 = list1->next;
+		if (file1->is_iterative == 1) {
+			/* Discard iterative manifests that are not from the current version */
+			if (file1->from_version == current_version) {
```

Comment:
I think you are failing to track manifests that have an iterative version that does not match the current_version since the version check is nested. Wouldn't a better check be
```C
if (file1->is_iterative == 1 && file1->from_version == current_version) {
        iterative_manifests = list_prepend_data(iterative_manifests, file1);
} else {
        full_manifests = list_prepend_data(full_manifests, file1);
}
```

Reply:
I'll update that too